salt/states/network.py

```diff
@@ -595,30 +595,16 @@ def managed(name, enabled=True, **kwargs):
 
     # Bring up/shutdown interface
     try:
-        # Get Interface current status
+        # Get Interfaces information
         interfaces = salt.utils.network.interfaces()
-        interface_status = False
-        if name in interfaces:
-            interface_status = interfaces[name].get("up")
+
+        # Check the interface current status
+        if name.split(':')[0] in interfaces:
+            interface_status = interfaces[name.split(':')[0]].get('up')
         else:
-            for iface in interfaces:
```

Comment:
Can you elaborate on why you are removing this section?

Reply:
Let's see if I still remember that :sweat_smile: 

So, although we get that nested result in the network.interfaces call, yet, when managing it with network.managed the name will always be either {interface}:{X} for the alias or any other simple name that does not match the main interface name, which AFAIR would be considered a different interface.

With that in mind, there is no real need to go through whatever is nested on an alias, as the main interface will either be up or down and the same would apply to the rest that is not nested. Otherwise (if it is not an alias to an interface and not an interface itself), it is probably not configured and hence down.

Reply:
Sounds good. Looks like we just need the changelog and test coverage to get this over the line

Reply:
I've spent some time reading about the test part and tried writing some new and improving what was present. It runs fine for me, but probably worth a second look. Also, changelog extension updated (included .md).
```
==================================================================================== test session starts ====================================================================================
platform linux -- Python 3.9.2, pytest-7.2.0, pluggy-0.13.0 -- /repo/.nox/test-parametrized-3-crypto-none-transport-zeromq-coverage-false/bin/python
cachedir: .pytest_cache
tempdir: /tmp/stsuite
max open files; soft: 3072; hard: 1048576
rootdir: /repo, configfile: pytest.ini
plugins: subtests-0.4.0, skip-markers-1.2.0, custom-exit-code-0.3.0, helpers-namespace-2021.4.29, shell-utilities-1.6.0, system-statistics-1.0.2, timeout-1.4.2, flaky-3.7.0, httpserver-1.0.4, tempdir-2019.10.12, salt-factories-1.0.0rc21
collected 7 items                                                                                                                                                                           

tests/integration/states/test_network.py::NetworkTest::test_managed_addInterface0 PASSED
tests/integration/states/test_network.py::NetworkTest::test_managed_addInterface1 PASSED
tests/integration/states/test_network.py::NetworkTest::test_managed_existingInterfaceNoChanges0 PASSED
tests/integration/states/test_network.py::NetworkTest::test_managed_existingInterfaceWithChanges0 PASSED
tests/integration/states/test_network.py::NetworkTest::test_managed_existingInterfaceWithChanges1 PASSED
tests/integration/states/test_network.py::NetworkTest::test_routes PASSED
tests/integration/states/test_network.py::NetworkTest::test_system PASSED

===================================================================================== warnings summary ======================================================================================
tests/integration/states/test_network.py::NetworkTest::test_managed_addInterface0
  /repo/.nox/test-parametrized-3-crypto-none-transport-zeromq-coverage-false/lib/python3.9/site-packages/napalm/base/helpers.py:8: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.10 it will stop working
    from collections import Iterable

tests/integration/states/test_network.py::NetworkTest::test_managed_addInterface0
  /repo/.nox/test-parametrized-3-crypto-none-transport-zeromq-coverage-false/lib/python3.9/site-packages/jnpr/junos/factcache.py:10: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.10 it will stop working
    class _FactCache(collections.MutableMapping):

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================================================================== 7 passed, 2 warnings in 33.13s ===============================================================================
```
Let me know if there is anything else from me here and thanks for the help! :+1: :smile: